sklearn/svm/base.py

```diff
@@ -482,8 +482,8 @@ def _validate_for_predict(self, X):
     @property
     def coef_(self):
         if self.kernel != 'linear':
-            raise ValueError('coef_ is only available when using a '
-                             'linear kernel')
+            raise AttributeError('coef_ is only available when using a '
```

Comment:
Note this is the only case that I can see where `ParameterError` would be used. As @jnothman said above, I wouldn't expect this particular error (trying to access `coef_` when `kerner != 'linear'`) to be handled programatically, and if you do it's straightforward to adapt your code in a not so invasive manner by using `except (ValueError, TypeError)`.